### ISSUE TITLE
DECODE-FLOAT returns single float significand for double float argument

### ISSUE THREAD

**Issue by robert-dodier** [armedbear/abcl#95]
CLHS says "decode-float computes three values that characterize float. The first value is of the same type as float and represents the significand."

I find that DECODE-FLOAT applied to double float arguments always returns a single float significand instead of a double float.

```
CL-USER(5): (type-of 1d0)        
DOUBLE-FLOAT
CL-USER(6): (type-of (decode-float 1d0))
SINGLE-FLOAT
CL-USER(7): (type-of most-positive-double-float)
DOUBLE-FLOAT
CL-USER(8): (type-of (decode-float most-positive-double-float))
SINGLE-FLOAT
CL-USER(9): (type-of least-positive-double-float)
DOUBLE-FLOAT
CL-USER(10): (type-of (decode-float least-positive-double-float))
SINGLE-FLOAT
```
I am working with ABCL built from svn as of r15125 if I am not mistaken. `lisp-implementation-version` reports: 
```
"1.6.0-dev"
"Java_HotSpot(TM)_Server_VM-Oracle_Corporation-1.8.0_60-b27"
"i386-Linux-3.13.0-44-generic"
```
```
CL-USER(6): *features*
(:X86 :JAVA-1.8 :UNIX :LINUX :ARMEDBEAR :ABCL :COMMON-LISP :ANSI-CL :CDR6 :MOP :PACKAGE-LOCAL-NICKNAMES)
```
`uname -a` reports: `Linux freekbox 3.13.0-44-generic #73-Ubuntu SMP Tue Dec 16 00:23:46 UTC 2014 i686 i686 i686 GNU/Linux`

**Comment by robert-dodier** [comment]
Here is a patch to fix this problem and also #93.

[decode-float-single-double-patch.txt](https://github.com/armedbear/abcl/files/2849949/decode-float-single-double-patch.txt)

**Comment by easye** [comment]
@robert-dodier Thanks for the diagnosis and the patch.  I must confess that I need to read up on the `decode-float` interface to take a stab at what is going on with the de-normalized float problems you identify in #94.
